src/forms/dispense-form.component.tsx

```diff
@@ -44,101 +41,82 @@ const DispenseForm: React.FC<DispenseFormProps> = ({
     if (!isSubmitting) {
       setIsSubmitting(true);
       const abortController = new AbortController();
-      medicationDispensesPayload.map((dispenseRequest) => {
-        saveMedicationDispense(dispenseRequest, abortController).then(
-          ({ status }) => {
-            if (status === 201 || status === 200) {
-              closeOverlay();
-              mutate();
-              showToast({
-                critical: true,
-                kind: "success",
-                description: t(
-                  "medicationListUpdated",
-                  "Medication dispense list has been updated."
-                ),
-                title: t(
-                  mode === "enter"
-                    ? "medicationDispensed"
-                    : "medicationDispenseUpdated",
-                  mode === "enter"
-                    ? "Medication successfully dispensed."
-                    : "Dispense record successfully updated."
-                ),
-              });
-            }
-          },
-          (error) => {
-            showNotification({
+      saveMedicationDispense(medicationDispensePayload, abortController).then(
```

Comment:
for example, here we only have to save a single medication dispense, not a set of them...